_posts/tutorials/2017-11-17-javalin-docker-example.md

```diff
@@ -0,0 +1,173 @@
+        This gives a significant performance boost and reduces the size of the application.
+        &mdash; <a href="https://opensource.com/resources/what-docker">opensource.com</a>
+    </p>
+</blockquote>
+
+## Initial Setup
+Before we get started, there are a few things we need to do:
+
+* Set up Docker [(Install Docker)](https://docs.docker.com/engine/installation/)
+* Deploy Registry server [(Deploy Registry)](https://docs.docker.com/registry/deploying/)
+* Install [Maven](https://maven.apache.org/guides/getting-started/maven-in-five-minutes.html)
+* Set up the Javalin Hello World example with Maven [(→ Tutorial)](/tutorials/maven-setup)
+
+## Configuring Maven
+This is actually where most of the work is done. In order to easily
+deploy a Java application anywhere, you have to create a jar file
+containing your application and all of its dependencies.
+Open the pom.xml of your Javalin Maven project and add the
+following configuration (below your dependencies tag):
+
+~~~markup
+<distributionManagement>
+        <repository>
+            <id>repo</id>
+            <name>Internal release</name>
+            <url>your-snapshot-repo-url</url>
+        </repository>
+</distributionManagement>
+
+<build>
+    <plugins>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <version>2.3.2</version>
+            <configuration>
+                <source>1.8</source>
+                <target>1.8</target>
+            </configuration>
+        </plugin>
+        <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-shade-plugin</artifactId>
+            <version>3.1.0</version>
+            <executions>
+                <!-- Run shade goal on package phase -->
+                <execution>
+                    <phase>package</phase>
+                    <goals>
+                        <goal>shade</goal>
+                    </goals>
+                    <configuration>
+                        <shadedArtifactAttached>true</shadedArtifactAttached>
+                        <transformers>
+                            <!-- add Main-Class to manifest file -->
+                            <transformer
+                                    implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
+                                <mainClass>HelloWorld</mainClass>
+                            </transformer>
+                        </transformers>
+                    </configuration>
+                </execution>
+            </executions>
+        </plugin>
+    </plugins>
+</build>
+~~~
+
+## Configuring Docker
+Before we can configure anything we must create a Dockerfile.
+we can create a text file using any editor and name it https://github.com/prasad-marne/javalin-docker/blob/master/Dockerfile 
```

Comment:
Something is wrong here. It shows up in the tutorial as:

>Before we can configure anything we must create a Dockerfile. we can create a text file using any editor and name it https://github.com/prasad-marne/javalin-docker/blob/master/Dockerfile Copy below contents to the file and move this file to root of your project.

Reply:
i see. let me fix this